packages/core-components/src/components/LogViewer/RealLogViewer.tsx

```diff
@@ -42,18 +43,26 @@ export function RealLogViewer(props: RealLogViewerProps) {
 
   const search = useLogViewerSearch(lines);
   const selection = useLogViewerSelection(lines);
+  const location = useLocation();
 
   useEffect(() => {
     if (search.resultLine !== undefined && listRef.current) {
       listRef.current.scrollToItem(search.resultLine - 1, 'center');
     }
   }, [search.resultLine]);
 
+  useEffect(() => {
+    if (location.hash) {
+      // #line-6 -> 6
+      const line = parseInt(location.hash.replace(/\D/g, ''), 10);
+      selection.setSelection(line, false);
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
   const handleSelectLine = (
     line: number,
     event: { shiftKey: boolean; preventDefault: () => void },
   ) => {
-    event.preventDefault();
```

Comment:
I removed this because it was preventing the url from updating. Not sure what use case was to have this here, but if it is important I can put it back and just update the URL with vanilla JS.